dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -83,53 +103,88 @@ def __add__(self, other: CategoricalColumn) -> CategoricalColumn:
         self._merge_calculations(
             merged_profile.__calculations, self.__calculations, other.__calculations
         )
-        # If both profiles have not met stop condition
-        if not (self._stop_condition_is_met or other._stop_condition_is_met):
-            merged_profile._categories = utils.add_nested_dictionaries(
-                self._categories, other._categories
+
+        if self.cms and other.cms:
+
+            assert isinstance(self._cms_max_num_heavy_hitters, int)
+            assert isinstance(other._cms_max_num_heavy_hitters, int)
+            cms_max_num_heavy_hitters: int = min(
```

Comment:
i'd say we could have assigned here and not output at merge (currently just passing through), but that can be a future refactor.